### PR TITLE
Upgrade the Symfony contracts

### PR DESCRIPTION
**Pull request by leofeyer**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -90,8 +90,8 @@
         "phpspec/php-diff": "^1.0",
         "phpunit/phpunit": "^9.5.19",
         "psr/cache": "^1.0 || ^2.0 || ^3.0",
-        "psr/container": "^1.0",
-        "psr/http-message": "^1.0",
+        "psr/container": "^1.0 || ^2.0",
+        "psr/http-message": "^1.0 || ^2.0",
```
Comment (leofeyer) [inline composer.json]: @ausi Does it make sense to keep all the old versions here? For example, `psr/container` version 1 is not installable with `symfony/service-contracts` 3 anyway.

Comment (ausi) [reply composer.json]: If it is not installable anyways, I’d remove the old version then to keep it clear and simple.

```json
        "psr/container": "^2.0",
        "psr/http-message": "^2.0",
```

**Review by ausi — DISMISSED** [review]

File: composer.json [hunk 2]
```diff
@@ -79,7 +79,7 @@
         "lcobucci/jwt": "^4.0",
         "league/commonmark": "^2.2",
         "league/flysystem": "^3.0",
-        "league/flysystem-bundle": "^2.0 || ^3.0",
+        "league/flysystem-bundle": "^3.0",
```
Comment (leofeyer) [inline composer.json:82]: Because version 2 is not installable with `league/flysystem` version 3.

File: composer.json [hunk 3]
```diff
@@ -89,10 +89,10 @@
         "paragonie/constant_time_encoding": "^2.2",
         "phpspec/php-diff": "^1.0",
         "phpunit/phpunit": "^9.5.19",
-        "psr/cache": "^1.0 || ^2.0 || ^3.0",
-        "psr/container": "^1.0",
-        "psr/http-message": "^1.0",
-        "psr/log": "^1.0 || ^2.0 || ^3.0",
+        "psr/cache": "^3.0",
```
Comment (leofeyer) [inline composer.json:92]: Because `symfony/cache-contracts` version 3 is not compatible with version 1 and 2.

File: composer.json [hunk 4]
```diff
@@ -89,10 +89,10 @@
...
+        "psr/container": "^2.0",
```
Comment (leofeyer) [inline composer.json:93]: Because `symfony/service-contracts` version 3 is not compatible with version 1.

File: composer.json [hunk 5]
```diff
@@ -89,10 +89,10 @@
...
+        "psr/http-message": "^1.0 || ^2.0",
```
Comment (leofeyer) [inline composer.json:94]: We could drop version 1 once https://github.com/terminal42/escargot/pull/33 has been merged, however, there are no packages preventing the installation of version 1, so we might as well keep it. @ausi WDYT?

File: composer.json [hunk 6]
```diff
@@ -89,10 +89,10 @@
...
+        "psr/log": "^1.1 || 2.0 || ^3.0",
```
Comment (leofeyer) [inline composer.json:95]: Version 1.0 is not installable anymore, but all other versions are.

File: composer.json [hunk 7]
```diff
@@ -160,7 +160,7 @@
         "symfony/yaml": "^6.3",
         "terminal42/escargot": "^1.4.1",
         "terminal42/service-annotation-bundle": "^1.1",
-        "toflar/psr6-symfony-http-cache-store": "^3.0 || ^4.0",
+        "toflar/psr6-symfony-http-cache-store": "^4.0",
```
Comment (leofeyer) [inline composer.json:163]: Version 3 is not compatible with Symfony 6, therefore it is not installable anymore.

File: composer.json [hunk 8]
```diff
@@ -194,9 +194,7 @@
         "contao/manager-plugin": "<2.0 || >=3.0",
         "doctrine/annotations": "<1.9",
         "doctrine/cache": "<1.10",
-        "friendsofsymfony/http-cache": "2.14.0",
```
Comment (leofeyer) [inline composer.json:197]: Our requirement is `^2.15.1`, so this will never match.

File: composer.json [hunk 9]
```diff
@@ -194,9 +194,7 @@
...
         "nikic/php-parser": "4.7.0",
-        "phpunit/phpunit": "<8.0",
```
Comment (leofeyer) [inline composer.json:199]: Our requirement is `^9.5.19`, so this will never match.

**Pull request merged**